pkg/cloudprovider/providers/azure/controllerCommon.go

```diff
@@ -0,0 +1,383 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package azure
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"hash/crc32"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	kwait "k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/golang/glog"
+)
+
+type armVmVhdDiskInfo struct {
+	Uri string `json:"uri"`
+}
+type armVmManagedDiskInfo struct {
+	Id string `json:"id"`
+}
+
+type armVmDataDisk struct {
+	Lun          int                   `json:"lun"`
+	Name         string                `json:"name,omitempty"`
+	CreateOption string                `json:"createOption"`
+	ManagedDisk  *armVmManagedDiskInfo `json:"managedDisk,omitempty"`
+	Vhd          *armVmVhdDiskInfo     `json:"vhd,omitempty"`
+	Caching      string                `json:"caching"`
+	DiskSizeGB   int                   `json:"diskSizeGB,omitempty"`
+}
+
+const (
+	aadTokenEndPointPath string = "%s/oauth2/token/"
+	apiversion           string = "2016-04-30-preview"
+	//_apiversionVhdVm      string = "2016-03-30"
+	diskEndPointTemplate string = "%ssubscriptions/%s/resourcegroups/%s/providers/microsoft.compute/disks/%s?api-version=%s"
+	vMEndPointTemplate   string = "%ssubscriptions/%s/resourcegroups/%s/providers/microsoft.compute/virtualmachines/%s?api-version=%s"
+	diskIdTemplate       string = "/subscriptions/%s/resourcegroups/%s/providers/microsoft.compute/disks/%s"
+	defaultDataDiskCount int    = 16 // which will allow you to work with most medium size VMs (if not found in map)
+
+	vhdBlobUriTemplate             = "https://%s.blob.core.windows.net/%s/%s"
```

Comment:
This is not accurate. You have to choose among `core.windows.net`, `core.usgovcloudapi.net`, etc....